datastream-common/src/main/java/com/linkedin/datastream/common/BrooklinEnvelope.java

```diff
@@ -75,6 +92,20 @@ public void setPreviousValue(Object previousValue) {
     _previousValue = previousValue instanceof Optional ? ((Optional<?>) previousValue).orElse(null) : previousValue;
   }
 
+  /**
+   * Get the Kafka headers
+   */
+  public Iterable<Header> getHeaders() {
```

Comment:
if we do change this to store `Headers` directly, we can return `Headers` here too.